src/coreclr/src/vm/genanalysis.cpp

```diff
@@ -18,27 +18,39 @@ uint32_t gcGenAnalysisBufferMB = 0;
 #ifndef GEN_ANALYSIS_STRESS
     if (gcGenAnalysisConfigured == GcGenAnalysisState::Uninitialized)
     {
-        if (CLRConfig::IsConfigOptionSpecified(W("GCGenAnalysisGen")))
+        bool match = true;
+        CLRConfigStringHolder gcGenAnalysisCmd(CLRConfig::GetConfigValue(CLRConfig::INTERNAL_GCGenAnalysisCmd));
+        if (gcGenAnalysisCmd != nullptr)
         {
+            // Get the managed command line.
+            LPCWSTR pCmdLine = GetCommandLineForDiagnostics();
+            size_t gcGenAnalysisCmdLength = wcslen(gcGenAnalysisCmd);
+            size_t cmdLineLength = wcslen(pCmdLine);
+            match = gcGenAnalysisCmdLength > cmdLineLength ? false : wcsncmp(gcGenAnalysisCmd, pCmdLine, gcGenAnalysisCmdLength) == 0;
```

Comment:
>match = gcGenAnalysisCmdLength > cmdLineLength ? false : wcsncmp(gcGenAnalysisCmd, pCmdLine, gcGenAnalysisCmdLength) == 0; [](start = 12, length = 122)

this seems very strict - not only do you have to specify the full cmdline, you even need to get the casing right. what do you think of something more like 

set complus_GCGenAnalysisCmd=w3wp-myapppool

? if the process name matches w3wp, myapppool just needs to appear somewhere in the cmdline. we have a regex impl (that I see eeconfig.cpp uses)

Reply:
It appears that the regular expression support in `EEConfig.h` is available only for debug builds, not something we can use.